### ISSUE TITLE
CSS/JS resources should always use proper load functions

### ISSUE THREAD

**Issue by Arantor** [SimpleMachines/SMF#1392]
Two reasons:
1) Means they get logged properly for the debugging information
2) Means that come release time, no-one has to go through and change hundreds of instances of ?alp21 to ?beta or whatever. At least one of the loaders already has facilities for including this cache buster.

Proper use will also cover off http://dev.simplemachines.org/mantis/view.php?id=3268

**Comment by Arantor** [comment]
Since it was brought to my attention that I wasn't detailed enough, and even though I took the trouble to see what others had added before me, I guess I can summarise this for everyone else. Even though these functions are in use in the source already.

loadCSSFile is the first, which accepts a filename as its simplest case, but accepts a variety of options in an array as the second parameter, these are all outlined in Load.php where the function is defined. An id can also be provided for caching purposes but is not explicitly required.

loadJavascriptFile is the second. Like loadCSSFile it accepts a filename as the simplest case which will normally be relative to the theme's directory. Again there is an array of options as the second parameter and an id for third. Of the options, 'default_theme' => true is the most useful, forcing the script to be loaded from the default theme. Again, the list of options is outlined in Load.php where the function is defined.

**Comment by MissAllSunday** [comment]
It was me, apparently the IRC staff only isn't that "staff-only anymore"...  it was made out of frustration, I already figure it out how it works. Not a big deal. Ask for logs to know exactly what I wrote to avoid further misunderstandings.

**Comment by Arantor** [comment]
Oh, I saw the quote. Fairly hard to misinterpret it. If you're going to make snide remarks about me, do it to my face. Mind you, I thought it was fairly hard to misinterpret "I don't care how fast anyone works as long as stuff gets done", but I guess we're both wrong.

Also, when the login form is loaded AJAXively, template layers don't get run, meaning that template_javascript() never gets run...

**Comment by MissAllSunday** [comment]
It was not about you, it was in general, if it was other person who wrote the report I would have said exactly the same thing. In fact, I thought exactly the same about other report I was looking at.

Once again, please ask for logs, I didn't wrote this: "I don't care how fast anyone works as long as stuff gets done"  so whoever told you or show you the quote is modifying the source, please ask for logs to avoid further misunderstandings. Thank you.

**Comment by Arantor** [comment]
-sigh- I know you didn't write that quote. BECAUSE I DID IN RESPONSE TO YOU (multiple times, in the team boards and now here). You have made it clear you think I had an issue with you not working at my speed. I DO NOT AND NEVER HAVE. Stop projecting onto what I'm saying.

You said about how you now had to spend time finding the 'proper functions' and 'then people get mad because I don't work as fast as they'. There's only one person that could possibly be in reference to. I say again: I do not, didn't and will not care that you don't have the same volume of output as I do. As I said, I can simply put that much more time into things so I'm bound to make more code. At the time I wrote that report, I envisaged me fixing it and it was more a case of recording what had to be done, rather than leaving enough details for other people. If I had thought I wouldn't be the one doing it, I would have left notes to use loadCSSFile and loadJavascriptFile in the report.

This is why I say there's a language barrier because you're still misinterpreting what I'm saying and I'm trying to say it as clearly as I can - it's almost like you're trying to find reasons to feel like a victim. You're better than that.

**Comment by MissAllSunday** [comment]
About the quote, my only concern was that you weren't misinformed, I don't know how you got that and I don't care either. I know you said it, I wasn't sure if your "source" added it to what I said, I have no idea what exactly you got, perhaps it was more than what I really said or was exaggerated or changed. My only concern was to make sure you got exactly what I said, no more, no less. Theres no victimization there.

I understood what you said, although, you gotta be honest too, your very first message wasn't exactly the same you end up saying at the end. Which is part of the reason people not always understand what you said.

Again, I wasn't thinking about you when I said that quote nor I was still thinking about that issue, it was a general rant just like the many many others I made and I will continue to do so, there wasn't any malice on it, it wasn't directed at you, I was frustrated after looking at the code for a long time without getting anywhere and the easiest thing for me to do to vent it was blaming the report, I repeat, the report, no you. I used the report as an scapegoat.

It was an easy way to vent and it worked since I was able to figure it out just  a few minutes later. Again, it wasn't anything personal, it was an extremely bad coincidence that you were the one who made the report and I know I made a mistake for making such a comment. For that I apologize, the next time I'm having issues or feeling frustrated I will do something different like punching a teddy bear or swearing on the roof of my building, whatever that doesn't cause any conflict.
